tests/test_pipeline.py

```diff
@@ -217,26 +211,19 @@ def test_defining_reusable_component_op_with_custom_spec():
         },
     )
 
-    load_from_hub_op_default_spec = ComponentSpec(
-        yaml_file_to_dict(load_from_hub_op_default_op.component_spec_path),
-    )
-
-    load_from_hub_op_custom_op = ComponentOp.from_registry(
-        name="load_from_hf_hub",
-        component_spec_path=custom_spec_path,
+    load_from_hub_custom_op = ComponentOp(
+        component_dir=load_from_hub_default_op.component_dir,
         arguments={
             "dataset_name": "test_dataset",
             "column_name_mapping": {"foo": "bar"},
             "image_column_names": None,
         },
     )
 
-    load_from_hub_op_custom_spec = ComponentSpec(yaml_file_to_dict(custom_spec_path))
-
-    assert load_from_hub_op_custom_op.component_spec == load_from_hub_op_custom_spec
-    assert load_from_hub_op_default_op.component_spec == load_from_hub_op_default_spec
-    assert load_from_hub_op_default_op.component_spec != load_from_hub_op_custom_spec
-    assert load_from_hub_op_custom_op.component_spec != load_from_hub_op_default_spec
+    assert (
+        load_from_hub_custom_op.component_spec
+        == load_from_hub_default_op.component_spec
+    )
```

Comment:
Some of these checks were trivial but harder to replicate, so only kept the one that is non-trivial.